### PR TITLE
Fix least squares options (closes #282)

### LINKED ISSUE #282: Some options are ignored for scipy least squares optimizers
Seems like `max_nfev` and some other options that can be set for the least squares optimizers in Scipy are ignored by pyPESTO. Would be good to be able to set these options (or at least print a warning, that it is ignored).

### PR DESCRIPTION
**Pull request by LeonardSchmiester**

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ pypesto/optimize/optimizer.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/optimizer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/optimizer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

File: pypesto/optimize/optimizer.py [hunk 1]
```diff
@@ -265,19 +266,21 @@ def minimize(
             jac = objective.get_sres if objective.has_sres else '2-point'
             # TODO: pass jac computing methods in options
 
+            if self.options is not None:
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:269]: can also move this to `__init__`

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/optimizer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/optimizer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/optimizer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/optimize/optimizer.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5375241&bid=17622680)

**Pull request merged**